packages/flutter/lib/src/widgets/focus_manager.dart

```diff
@@ -1677,8 +1677,13 @@ class FocusManager with DiagnosticableTreeMixin, ChangeNotifier {
     if (kFlutterMemoryAllocationsEnabled) {
       ChangeNotifier.maybeDispatchObjectCreation(this);
     }
-    _appLifecycleListener = _AppLifecycleListener(_appLifecycleChange);
-    WidgetsBinding.instance.addObserver(_appLifecycleListener);
+    if (kIsWeb || defaultTargetPlatform != TargetPlatform.android) {
+      // It appears that some Android keyboard implementations can cause
+      // app lifecycle state changes: adding this listener would cause the
+      // text field to unfocus as the user is trying to type.
```

Comment:
Nit: Maybe mention that this is a temporary fix and/or link to https://github.com/flutter/flutter/pull/142930?